packages/remix/src/ssr/rootAuthLoader.ts

```diff
@@ -14,11 +14,11 @@ import {
 } from './utils';
 
 interface RootAuthLoader {
-  <Options extends RootAuthLoaderOptions>(
+  <Callback extends RootAuthLoaderCallback<Options>, Options extends RootAuthLoaderOptions>(
```

Comment:
❓ Isn't this a breaking change ? For users that have strictly-typed the usage of `rootAuthLoader`

Reply:
@panteliselef Could you provide an example?

This used to return a `LoaderFunctionReturn` type as shown here:
```ts
interface LoaderFunction {
    (args: DataFunctionArgs): Promise<Response> | Response | Promise<AppData> | AppData;
}

export type LoaderFunctionReturn = ReturnType<LoaderFunction>;
```

Which I think isn't especially useful so I doubt anyone is using this directly. 

On the contrary, the `LoaderFunction` should be generic enough to accept the new type if used as such: 
```ts
export const loader: LoaderFunction = args => {
  return rootAuthLoader(args, ({ request }) => {
    const { user } = request;
    return { message: 'hello' };
  });
};
```

Sorry if I totally misunderstood your point!

Reply:
@nikosdouvlis Sorry for not providing an example earlier.

Based on the changes made by this PR, wouldn't the following code throw an error because the first type argument of the generic does not match `Callback extends RootAuthLoaderCallback<Options>` ?
```typescript
export const loader: LoaderFunction = args => {
  return rootAuthLoader<{loadUser: true}>(
    args,
    ({ request }) => {
      const { user } = request;
      console.log('User:', request);
      return { user };
    },
    {
      loadUser: true
    },
  );
};
```

Reply:
@panteliselef thank you for the comment. I've updated the Generics order.